### DISCUSSION TITLE
Coolify support?!

### DISCUSSION THREAD

**Discussion by thereis** [rahulhaque/soketi-app-manager-filament#8] [Ideas]
Would be really cool to see this project integrated with coolify, since they already provide a solution for soketi!

**Comment by rahulhaque** [comment]
@thereis I'm curious! What type of solution coolify provides? What kind of integration are you talking about?

**Reply by thereis** [reply]
First of all, thanks for replying me back. Coolify it's an open source service where you can manage your server resources and docker applications in the cloud or self hosted. It's a pretty convenient tool for server owners to deploy apps, it has been trending in the past few months.

Since it can deploy apps in a click, it would be a very cool to see your application there integrating with their soketi installation.

Feel free to take a look:

https://coolify.io

**Reply by rahulhaque** [reply]
Ok, now I get it. You would like to see this project as a one click installable service in coolify. Let me see what I can come up with.

**Reply by thereis** [reply]
@rahulhaque any progress so far?

**Reply by rahulhaque** [reply]
@thereis not any unfortunately. I am eager but busy.

**Reply by thereis** [reply]
@rahulhaque would you like a bounty to get an extra stamina? 😄

**Reply by rahulhaque** [reply]
@thereis lol not at all 😄. Thanks for asking though. I do open-source staffs for free. Bounty puts pressure similar to a job. I cannot rest well till I've done my part. I have a [ko-fi](https://ko-fi.com/rahulhaque) account running just for people who used, got benefited and thinks my works are worthy of a donation.

Now, about what you were asking, did you try running the `docker-compose.yml` from my app-manager in coolify? How did it turned out?

Another thing I noticed is that coolify doesn't have any one-click service named soketi, see [here](https://coolify.io/docs/services). It ships with its own soketi implementation for its own needs. Did you ask to manage that soketi server with my app-manager?

**Reply by thereis** [reply]
@rahulhaque, hey! I thought placing a reward would motivate you and also would help the community of having this as 1-click service as well 😄 but I totally understand and respect your point.

Regarding the `docker-compose.yml` file, I've tried myself but it's not working completely right, it doesn't even install the composer libraries and since I am not a PHP developer, I have no such expertise with Laravel and it's dependencies, so it's been very tricky :(

I was intending to use your management tool to admin a current working soketi, but since it's a really simple application, I don't have any other problems of migrating to the one deployed into the composer stack.

Regarding soketi being 1-click app, you can see it here:

![image](https://github.com/user-attachments/assets/85036424-3616-4af9-9edd-36b87574ea59)

It's inside of their panel, so we might assume the service list in docs it's not updated 😢

**Reply by rahulhaque** [reply]
@thereis thanks for the explanation. As you've mentioned you're not a PHP developer, I feel the need for a standalone version of my app even more.

My current focus is setuping coolify in my local machine. If I can run it locally without any issue, I'll be able to come up with a solution fast. If successful and if this helps you, only then you can make a donation if you want.

Let me see if I can get it done.

**Reply by thereis** [reply]
@rahulhaque looking forward to hear more about it 😊

**Reply by rahulhaque** [reply]
@thereis can you please add a new git based application in coolify with this repo, select build pack > Dockerfile, deploy and let me know of the outcome? I have added an standalone dockerfile temporarily in `main` branch because of [this](https://github.com/coollabsio/coolify/issues/4584) bug.

**Reply by rahulhaque** [reply]
@thereis I'll be adding a one click installer soon (if everything goes right). The installer will only contain the app-manager as coolify provides database and soketi as separate services. They will be connected through environment variables. Let's hope for the best.

Meanwhile, you try out the dockerfile which I was able to run successfully locally. Let me know if you're stuck.

**Reply by thereis** [reply]
Hey @rahulhaque thanks for your message, I was on a busy weekend, I will spare a time right now to figure out this issue, thanks for helping me out! I will keep you updated.

**Reply by thereis** [reply]
@rahulhaque should I load from the docker-compose file or from the nixpacks?

**Reply by rahulhaque** [reply]
@thereis thanks for the feedback. I've just pushed a docker-compose update. Yes, you should load from the docker-compose. It should work right out of the box. After first run go to terminal and run `php artisan key:generate --show`. Copy paste the key to APP_KEY environment variable. See how that goes.

**Reply by thereis** [reply]
hey @rahulhaque thank you, now I can see the authentication page but when I use the default credentials in the docs, there's a laravel error happening there... do I have to run any command to seed the database? 

also, does it support postgres? do I have to pass any env variable to link my database?

**Reply by rahulhaque** [reply]
@thereis yes, you have to set the required environment variables. I have included them [here](https://github.com/rahulhaque/soketi-app-manager-filament/blob/main/docker-compose.yaml) but don't know if they are visible in coolify UI. 

After you set up the environment variables, go to terminal like before and seed the database. Before that, make sure your database service is ready. Use coolify to spin up a database.

It does support postgres, just set `DB_CONNECTION=pgsql`.

Redis is used for caching. Set those values as well.

Add `APP_DEBUG=true` to see the underlying errors.

**Reply by thereis** [reply]
@rahulhaque  oh, cool, so, the only bootstrapped env var that have been automatically created it was `$COOLIFY_URL`, after cloning the repo and fetching the yaml file, this env var was the only that coolify created automatically for me.

I will proceed here to manually add the other env vars, but I am kindly not sure that it will work since it was not properly mapped...

**Reply by rahulhaque** [reply]
@thereis I think they will. I've added the variables just for the UI. Need to investigate why it didn't work. I'll wait for your result.

**Reply by thereis** [reply]
@rahulhaque is there a way to customize the users table? I do have one with the same naming so it's kinda conflicting with my current database hahaha, also, it looked good here setting the env vars from the coolify dashboard... even though they were not automatically created

**Reply by rahulhaque** [reply]
@thereis there's but I think you should use a separate database for soketi + soketi app manager. Your application database should be separate. This is by design. For example, your app and pusher are two separate service. They have separate user base.

**Reply by thereis** [reply]
thanks @rahulhaque, I will setup another database here and I will update you

**Reply by rahulhaque** [reply]
@thereis also, I think I'll ship database with this as well if possible so that soketi service can make use of app-manager database. Seeding can be automated this way. Meanwhile can you please test if every feature is working, connect to soketi and provide feedback.

**Reply by thereis** [reply]
@rahulhaque tried here but it looks like the env var for `DB_CONNECTION` is not being replaced in the coolify file:

```
DB_CONNECTION: mysql <------- DIDNT WORKED
DB_HOST: [REDACTED IT WORKED]
DB_PORT: '3306' <------- DIDNT WORKED
DB_DATABASE: soketi_app_manager  <------- DIDNT WORKED
DB_USERNAME: root  <------- DIDNT WORKED
DB_PASSWORD: [REDACTED IT WORKED]
```

**Reply by rahulhaque** [reply]
@thereis did you use the generated database name, username, password from coolify service and still didn't work? 
```
DB_CONNECTION: mysql
DB_HOST: GENERATED_HOST_NAME
DB_PORT: 3306
DB_DATABASE: GENERATED_DB_NAME
DB_USERNAME: GENERATED_USERNAME
DB_PASSWORD: GENERATED_PASSWORD
```

**Reply by thereis** [reply]
@rahulhaque from the yaml file it didn't generated any database, only the app and one env var which was the COOLIFY_URL, which is not a env var if you use their cloud version. When I tried to override the env vars, from the configuration screen, it didn't worked because the values were hardcoded in the configuration there. I tried myself to enter in the container and then execute the seed command but the configuration was hardcoded there as well.

**Reply by rahulhaque** [reply]
@thereis it won't generate database which is normal. I've updated docker-compose file. Environment variable should be visible in the UI.

After you set up the database credentials, go terminal and run `php artisan migrate --seed`. If your db connection is successful it will create the database.

**Reply by rahulhaque** [reply]
@thereis you should be able to drop to terminal from the coolify UI. You can try the following -
```bash
# Copy .env.example to .env
# Change needed variables
cp .env.example .env

# Generate application key
php artisan key:generate

# Migrate database
php artisan migrate --seed
```
If this works, environment variables in coolify UI won't be needed and deleted. Let me know which worked better.

**Reply by thereis** [reply]
![image](https://github.com/user-attachments/assets/1c299995-540d-4a57-9782-ca4a05f314e2)

@rahulhaque awesome! Now the env vars are working correctly. The approach that I've done was pretty much to reload the yaml file and the env vars have been read correctly from coolify settings page. The database have been created correctly but no data have been seeded into it, the command that I've run was pretty much `php artisan migrate --seed`

Should I create the users manually and if so, what's the hash to salt the password?

**Reply by rahulhaque** [reply]
@thereis that's weird. It should've created the user. Well, go to terminal and run `php artisan make:filament-user --name <your_name> --email <your_email> --password <your_password>`.

**Reply by thereis** [reply]
@rahulhaque amazing, now I can see the dashboard, it doesn't have any stats yet because I didn't configured my soketi instance properly, but I will do that and notify you as soon as possible, thank you for you help!

**Reply by thereis** [reply]
@rahulhaque what `PUSHER_SCHEME` means?

**Reply by rahulhaque** [reply]
@thereis either `http` or `https`. See the `.env.example` for any reference.

**Reply by rahulhaque** [reply]
@thereis I've also updated the readme recently and added coolify installation part. Please have a look at the steps when you're free and suggest if any changes are necessary there.

**Reply by thereis** [reply]
@rahulhaque thanks! no updates needed, I will reach to coolify in order to find a way to export the metrics service because right now they only export the socket port.

I wish you a happy new year and thank you for your help!

**Reply by rahulhaque** [reply]
@thereis thanks a lot for the donation and loving my work 🙏 🤗 Your contribution helps the community grow. 🎉

Wish you a happy new year.

**Reply by rahulhaque** [reply]
@thereis if you happen to deploy this again in future, please follow the instructions in `master` branch and let me know of any issues. I will be testing in the `main` branch for other possibilities. Also, I think the new instructions in the `master` branch is a bit simpler. Thank you.
